### PR TITLE
fix for sync algorithm with open children of closed extensions

### PR DESCRIPTION
**Pull request by czue**
hopefully this is the actual fix for https://github.com/dimagi/commcare-hq/pull/14327

@proteusvacuum

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -986,10 +960,11 @@ def _delete_index(self, index):
 
     def update_phone_lists(self, xform, case_list):
         made_changes = False
-        logger.debug('updating sync log for {}'.format(self.user_id))
-        logger.debug('case ids before update: {}'.format(', '.join(self.case_ids_on_phone)))
-        logger.debug('dependent case ids before update: {}'.format(', '.join(self.dependent_case_ids_on_phone)))
-        logger.debug('index tree before update: {}'.format(self.index_tree))
+        _get_logger().debug('updating sync log for {}'.format(self.user_id))
+        _get_logger().debug('case ids before update: {}'.format(', '.join(self.case_ids_on_phone)))
+        _get_logger().debug('dependent case ids before update: {}'.format(', '.join(self.dependent_case_ids_on_phone)))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E501 line too long (119 > 115 characters)

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -1088,14 +1063,19 @@ def get_latest_owner_id(case_id, action=None):
...
+            _get_logger().debug('case {} is NOT live.'.format(update.case_id))
             if update.has_extension_indices_to_add():
                 # non-live cases with extension indices should be added and processed
                 self.case_ids_on_phone.add(update.case_id)
                 for index in update.indices_to_add:
                     self._add_index(index, update)
                     made_changes = True
 
+        _get_logger().debug('case ids mid update: {}'.format(', '.join(self.case_ids_on_phone)))
+        _get_logger().debug('dependent case ids mid update: {}'.format(', '.join(self.dependent_case_ids_on_phone)))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E501 line too long (116 > 115 characters)

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 3]
```diff
@@ -1108,14 +1088,14 @@ def get_latest_owner_id(case_id, action=None):
                         self._delete_index(index)
                 made_changes = True
 
-        logger.debug('case ids after update: {}'.format(', '.join(self.case_ids_on_phone)))
-        logger.debug('dependent case ids after update: {}'.format(', '.join(self.dependent_case_ids_on_phone)))
-        logger.debug('index tree after update: {}'.format(self.index_tree))
-        logger.debug('extension index tree after update: {}'.format(self.extension_index_tree))
+        _get_logger().debug('case ids after update: {}'.format(', '.join(self.case_ids_on_phone)))
+        _get_logger().debug('dependent case ids after update: {}'.format(', '.join(self.dependent_case_ids_on_phone)))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E501 line too long (118 > 115 characters)

File: corehq/ex-submodules/casexml/apps/phone/tests/test_index_tree.py [hunk 4]
```diff
@@ -445,6 +445,43 @@ def test_purge_extension_host_is_parent(self):
         self.assertTrue(child_id in sync_log.case_ids_on_phone)
         self.assertTrue(host_id in sync_log.case_ids_on_phone)
 
+    def test_open_extension_of_extension(self):
+        [host_id, extension_id, extension_of_extension_id] = all_ids = ['host', 'extension', 'extension_of_extension']
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_index_tree.py]: E501 line too long (118 > 115 characters)

**Comment by ctsims** [comment]
Hey @proteusvacuum and @czue , should we be adding these to the mobile tests as well?

**Comment by czue** [comment]
@ctsims https://github.com/dimagi/commcare-core/pull/498 - unless that's not what you're talking about?

**Comment by ctsims** [comment]
@czue Ah, gotcha. I missed it since it had already been pulled. thanks.

**Pull request merged**